### PR TITLE
fix: Do not set property if empty or null

### PR DESCRIPTION
**Pull request by kennethmyhra**
* If contents from flat file is non-existing, do not set the property on
  the FHIR object

### CODE REVIEW

File: src/fhir-tool/Program.cs [hunk 1]
```diff
@@ -452,34 +452,44 @@ private static IList<Questionnaire> GetQuestionnairesFromFlatFileFormatV2(string
...
+                if (!string.IsNullOrEmpty(masterDetail.Master.Status))
+                    questionnaire.Status = EnumUtility.ParseLiteral<PublicationStatus>(masterDetail.Master.Status);
+                if (!string.IsNullOrEmpty(masterDetail.Master.Date))
+                    questionnaire.Date = masterDetail.Master.Date;
+                if (!string.IsNullOrEmpty(masterDetail.Master.Publisher))
+                    questionnaire.Publisher = masterDetail.Master.Publisher;
+                if (!string.IsNullOrEmpty(masterDetail.Master.Description))
+                    questionnaire.Description = new Markdown(masterDetail.Master.Description);
+                if (!string.IsNullOrEmpty(masterDetail.Master.Purpose))
+                    questionnaire.Purpose = string.IsNullOrEmpty(masterDetail.Master.Purpose) ? null : new Markdown(masterDetail.Master.Purpose);
```
Comment (autark) [inline src/fhir-tool/Program.cs]: Duplicate check

File: src/fhir-tool/Program.cs [hunk 2]
```diff
@@ -607,17 +603,23 @@ private static Questionnaire.ItemComponent CreateItemComponentV2(QuestionnaireIt
...
+                itemComponent.Prefix = item.Prefix;
+            if (!string.IsNullOrEmpty(item.Text))
+                itemComponent.Text = item.Text;
 
             if (itemType != Questionnaire.QuestionnaireItemType.Group && itemType != Questionnaire.QuestionnaireItemType.Display)
             {
-                itemComponent.Required = item.Required;
+                if(item.Required.HasValue)
+                    itemComponent.Required = item.Required;
                 itemComponent.ReadOnly = item.ReadOnly;
```
Comment (autark) [inline src/fhir-tool/Program.cs]: No guard for "ReadOnly"?

File: src/fhir-tool/Program.cs [hunk 3]
```diff
@@ -731,44 +733,39 @@ private static IList<Questionnaire> GetQuestionnairesFromFlatFileFormatV1(string
...
+                if (!string.IsNullOrEmpty(masterDetail.Master.Status))
+                    questionnaire.Status = EnumUtility.ParseLiteral<PublicationStatus>(masterDetail.Master.Status);
+                if (!string.IsNullOrEmpty(masterDetail.Master.Date))
+                    questionnaire.Date = masterDetail.Master.Date;
+                if (!string.IsNullOrEmpty(masterDetail.Master.Publisher))
+                    questionnaire.Publisher = masterDetail.Master.Publisher;
+                if (!string.IsNullOrEmpty(masterDetail.Master.Description))
+                    questionnaire.Description = new Markdown(masterDetail.Master.Description);
+                if (!string.IsNullOrEmpty(masterDetail.Master.Purpose))
+                    questionnaire.Purpose = string.IsNullOrEmpty(masterDetail.Master.Purpose) ? null : new Markdown(masterDetail.Master.Purpose);
```
Comment (autark) [inline src/fhir-tool/Program.cs]: more duplicate checks

File: src/fhir-tool/Program.cs [hunk 4]
```diff
@@ -731,44 +733,39 @@ private static IList<Questionnaire> GetQuestionnairesFromFlatFileFormatV1(string
...
+                if (!string.IsNullOrEmpty(masterDetail.Master.Contact))
+                    questionnaire.Contact = new List<ContactDetail> { new ContactDetail { Name = masterDetail.Master.Contact } };
+
```
Comment (autark) [inline src/fhir-tool/Program.cs]: This has changed from previous version. Was:
Contact = new List<ContactDetail> { new ContactDetail { Telecom = new List<ContactPoint> { new ContactPoint { System = ContactPoint.ContactPointSystem.Url, Value = masterDetail.Master.Contact } } } }
Is the new version correct?

File: src/fhir-tool/Program.cs [hunk 5]
```diff
@@ -895,14 +892,28 @@ private static Questionnaire.ItemComponent CreateItemComponentV1(QuestionnaireIt
...
+            if (!string.IsNullOrEmpty(item.Prefix))
+                itemComponent.Prefix = item.Prefix;
+            if (!string.IsNullOrEmpty(item.Text))
+                itemComponent.Text = item.Text;
+
+            if (itemType != Questionnaire.QuestionnaireItemType.Group && itemType != Questionnaire.QuestionnaireItemType.Display)
+            {
+                if (item.Required.HasValue)
+                    itemComponent.Required = item.Required;
+                itemComponent.ReadOnly = item.ReadOnly;
```
Comment (autark) [inline src/fhir-tool/Program.cs]: no guard for this one as well. Intentional?

File: src/fhir-tool/Program.cs [hunk 6]
```diff
@@ -607,17 +603,23 @@ private static Questionnaire.ItemComponent CreateItemComponentV2(QuestionnaireIt
             Questionnaire.ItemComponent itemComponent = new Questionnaire.ItemComponent
             {
                 Type = itemType,
-                LinkId = item.LinkId,
-                Prefix = string.IsNullOrEmpty(item.Prefix) ? null : item.Prefix,
-                Text = string.IsNullOrEmpty(item.Text) ? null : item.Text
             };
+            if (!string.IsNullOrEmpty(item.LinkId))
```
Comment (autark) [inline src/fhir-tool/Program.cs]: Code is getting somewhat verbose with all the if's.
Would it be possible to retain the more terse syntax:

Property = string.IsNullOrEmpty(item.Property) ? null : item.Property

or does assigning null to properties have unwanted side effects?

File: src/fhir-tool/Program.cs (hunk 2 above)
Comment (kennethmyhra) [reply src/fhir-tool/Program.cs]: item.ReadOnly will always be either true or false, no need to guard it.

File: src/fhir-tool/Program.cs (hunk 5 above)
Comment (kennethmyhra) [reply src/fhir-tool/Program.cs]: Yep, reason same as above

File: src/fhir-tool/Program.cs (hunk 6 above)
Comment (kennethmyhra) [reply src/fhir-tool/Program.cs]: I agree that it gets quite verbose, but unwanted side effects is the reason currently

**Pull request merged**